File: 2015/02/02.kt
import java.io.File

fun part01(input: String): Int {
    // Extract Length, Width and Height
    val (l, w, h) = input
                    .split("x")
                    .map { it.toInt() }
                    .sortedDescending()
    
    return 2*l*w + 2*w*h + 2*h*l + w*h
}

fun part02(input: String): Int {
    val (l, w, h) = input
                    .split("x")
                    .map { it.toInt() }
                    .sortedDescending()
    
    return w+w+h+h + l*w*h
}

fun main() {
    val inputString = File("input.txt").readLines()
    assert(part01("2x3x4") == 58)
    assert(part01("1x1x10") == 43)
    
    assert(part02("2x3x4") == 34)
    assert(part02("1x1x10") == 14)

    var totalPaper = 0
    var totalRibbon = 0
    for (input in inputString) {
        totalPaper += part01(input)
        totalRibbon += part02(input)
    }
    println("Part 01: ${totalPaper}")
    println("Part 02: ${totalRibbon}")
}

File: 2015/02/02.rb

def part01(input)
    l, w, h = input
              .split("x")
              .map do |it| it.to_i end
              .sort
              .reverse
    
    return 2*l*w + 2*w*h + 2*h*l + w*h
end

def part02(input)
    l, w, h = input
              .split("x")
              .map do |it| it.to_i end
              .sort
              .reverse
    
    return w+w+h+h + l*w*h
end


File.open("input.txt") do |file|
    Inputs = file.readlines

    totalPaper = 0
    totalRibbon = 0
    Inputs.each do |input|
        totalPaper += part01(input)
        totalRibbon += part02(input)
    end
    puts "Part01: #{totalPaper}"
    puts "Part02: #{totalRibbon}"
end
